### PR TITLE
test: cover setup() paths and reach 100% coverage

### PR DESCRIPTION
**Pull request by bluetoothbot**
## What
Expand `tests/test_init.py` from 2 constructor-only tests to 11 tests covering the previously-untested `setup()` and `async_setup()` paths on both `BluetoothDevice` and `UARTDevice`.

## Why
The library was at 68% coverage with all the actual sysfs-parsing logic (lines 37, 41-43, 67, 71-79) untested. The `setup()` methods are where bugs would actually live; the constructors are trivial.

## How
- Uses `monkeypatch` to redirect `BLUETOOTH_DEVICE_PATH` / `UART_DEVICE_PATH` to a `tmp_path`-rooted fake sysfs.
- Writes realistic `uevent` content and a `device` symlink to exercise `Path.readlink()` and the `OF_COMPATIBLE_0` / `MODALIAS` parsers.
- Adds edge-case tests: missing fields, `MODALIAS` without a comma.
- Verifies `async_setup` actually offloads to `loop.run_in_executor`.

## Testing
`poetry run pytest --cov=uart_devices` → 11 passed, **100%** coverage (was 68%).

---
### Quality Report

**Changes**: 1 file changed, 158 insertions(+), 1 deletion(-)

**Code scan**: clean

**Tests**: passed (11 passed)

**Branch hygiene**: clean

*Generated by Kōan post-mission quality pipeline*

### CODE REVIEW

File: tests/test_init.py [hunk 1]
```diff
@@ -1,14 +1,171 @@
...
+    """async_setup must offload blocking I/O to the executor."""
+    monkeypatch.setattr("uart_devices.UART_DEVICE_PATH", tmp_path)
+    _write_uevent(tmp_path / "serial0-0", "OF_COMPATIBLE_0=brcm,bcm43438-bt\n")
+    device = UARTDevice("serial0-0")
+    device.path = tmp_path / "serial0-0"
+
+    async def main():
+        loop = asyncio.get_running_loop()
+        called_in_executor = {"value": False}
+        original_run_in_executor = loop.run_in_executor
```
Comment (bdraco) [inline tests/test_init.py]: this is hacky

**Review by bdraco — CHANGES_REQUESTED** [review]
fix run_in_executor hack to be proper pytest

**Comment by bdraco** [comment]
@bluetoothbot rr

File: tests/test_init.py (hunk 1 above)
Comment (bluetoothbot) [reply tests/test_init.py]: Agreed. The hackiness has three parts worth fixing together:

1. **Async harness mismatch** — every other async test in this file uses `@pytest.mark.asyncio`, but this one wraps an inner `main()` in `asyncio.run()`. That alone makes it stand out.
2. **`monkeypatch.setattr(loop, "run_in_executor", tracker)`** — patching a bound method on a live event-loop instance via pytest's monkeypatch is fragile and reads ad-hoc. Swapping in `unittest.mock.patch.object(loop, "run_in_executor", wraps=loop.run_in_executor)` gives you a real spy with `assert_called_once()` and proper cleanup via the context manager.
3. **Misleading name** — `test_bluetooth_device_setup_runs_in_executor_via_async` actually exercises `UARTDevice.async_setup`. Rename to `test_uart_device_async_setup_uses_executor`.

Suggested rewrite:

```python
@pytest.mark.asyncio
async def test_uart_device_async_setup_uses_executor(monkeypatch, tmp_path):
    monkeypatch.setattr("uart_devices.UART_DEVICE_PATH", tmp_path)
    _write_uevent(tmp_path / "serial0-0", "OF_COMPATIBLE_0=brcm,bcm43438-bt\n")
    device = UARTDevice("serial0-0")

    loop = asyncio.get_running_loop()
    with mock.patch.object(loop, "run_in_executor", wraps=loop.run_in_executor) as spy:
        await device.async_setup()

    spy.assert_called_once()
    assert device.manufacturer == "brcm"
```

Alternative: just delete this test. `test_uart_device_async_setup` already proves the coroutine populates the fields correctly, and "did it call `run_in_executor`?" is implementation detail rather than observable behaviour. Happy to go either way — your call.
